clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -97,17 +97,17 @@ public class ProducerConfig extends AbstractConfig {
                                                  + "batch size is under this <code>batch.size</code> setting.";
 
     /** <code>partitioner.adaptive.partitioning.enable</code> */
-    public static final String PARTITIONER_ADPATIVE_PARTITIONING_ENABLE_CONFIG = "partitioner.adaptive.partitioning.enable";
-    private static final String PARTITIONER_ADPATIVE_PARTITIONING_ENABLE_DOC =
-            "When set to 'true', the producer will try to adapt to broker performance and produce more messages to partitions hosted on faster brokers. "
+    public static final String PARTITIONER_ADAPTIVE_PARTITIONING_ENABLE_CONFIG = "partitioner.adaptive.partitioning.enable";
```

Comment:
Strictly speaking, this typo has formed part of the public interface since it was introduced in 2022. So, really, you should leave the existing constants in place and mark them as deprecated. They can then be removed in AK 5.0. Otherwise, any applications which happened to use the wrong spelling would fail to build.